File: WindowsCE/NurSample_vs2008/NurSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using NordicId;
using NurApiDotNet; //Need to use namespace of NurApiDotNet

namespace NurSample
{
    public partial class Form1 : Form
    {
        /// <summary>
        /// The NurApi handle
        /// </summary>
        NurApi hNur = null;                 // Handle of NurApi

        /// <summary>
        /// Initializes a new instance of the <see cref="Form1" /> class.
        /// </summary>
        public Form1()
        {
            InitializeComponent();
            Fullscreen.Init();
            Fullscreen.SetFullScreen(true);
            this.Text = string.Format("{0} v{1}",
                System.Reflection.Assembly.GetExecutingAssembly().GetName().Name,
                System.Reflection.Assembly.GetExecutingAssembly().GetName().Version);
            this.WindowState = FormWindowState.Maximized;

            try
            {
                // Call NurApi constructor and give Form object for receive notifications
                // in same thread where this Control is running
                hNur = new NurApi(this); //Handle of NurApi
                //hNur.SetLogLevel(NurApi.LOG_ALL & ~NurApi.LOG_DATA);
                NurCapabilities.I.Nur = hNur;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message, Program.appName);
                this.Close();
                return;
            }

            // Set NurApi for UserControls
            nurLog.SetNurApi(hNur);
            nurInfo.SetNurApi(hNur);
            nurInventory.SetNurApi(hNur);
            nurAntennas.SetNurApi(hNur);
            nurWriter.SetNurApi(hNur);
            nurLocator.SetNurApi(hNur);
            nurSettings.SetNurApi(hNur);
            nurNxp.SetNurApi(hNur);

            try
            {
                // Configure Scan & Trigger -buttons
                HHScanButton.ConfigureScanButtons((Keys)NordicId.VK.SCAN);
                HHScanButton.ScanButtonMode = HHScanButton.SCANMODE.NONE;
            }
            catch (Exception)
            {

            }
        }

        /// <summary>
        /// Handles the Load event of the Form1 control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs" /> instance containing the event data.</param>
        private void Form1_Load(object sender, EventArgs e)
        {
            // Connect Nur -module
            hNur.ConnectIntegratedReader();
        }

        /// <summary>
        /// Handles the Closing event of the Form1 control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="CancelEventArgs" /> instance containing the event data.</param>
        private void Form1_Closing(object sender, CancelEventArgs e)
        {
            // Restore Scan & Trigger -button configurations
            HHScanButton.RestoreScanButtons();

            // Stop locationg thread if running
            nurLocator.StopLocating();

            // Dispose NurApi
            if (hNur != null)
            {
                if (hNur.IsConnected())
                    hNur.Disconnect();
                hNur.Dispose();
            }
            Fullscreen.SetFullScreen(false);
        }

        private void Form1_KeyDown(object sender, KeyEventArgs e)
        {
            // Make sure that this.KeyPreview is turned on.
            switch (e.KeyCode)
            {
                case (Keys)NordicId.VK.SCAN:
                    // Jump to inventory -tab and start / stop inventory
                    tabControl1.SelectedIndex = tabControl1.TabPages.IndexOf(inventoryTab);
                    nurInventory.startStopInventoryBtn_Click(this, EventArgs.Empty);
                    e.Handled = true;
                    break;
            }
        }
    }
}

File: WindowsCE/NurSample_vs2008_VB/NurSampleCE_VB/Form1.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports NordicId
Imports NurApiDotNet
'Need to use namespace of NurApiDotNet
Partial Public Class Form1
    Inherits Form
    ''' <summary>
    ''' The NurApi handle
    ''' </summary>
    Private hNur As NurApi = Nothing
    ' Handle of NurApi
    ''' <summary>
    ''' Initializes a new instance of the <see cref="Form1" /> class.
    ''' </summary>
    Public Sub New()
        InitializeComponent()
        Fullscreen.Init()
        Fullscreen.SetFullScreen(True)
        Me.Text = String.Format("{0} v{1}", System.Reflection.Assembly.GetExecutingAssembly().GetName().Name, System.Reflection.Assembly.GetExecutingAssembly().GetName().Version)
        Me.WindowState = FormWindowState.Maximized

        Try
            ' Call NurApi constructor and give Form object for receive notifications
            ' in same thread where this Control is running
            hNur = New NurApi(Me)
            'Handle of NurApi
            'hNur.SetLogLevel(NurApi.LOG_ALL & ~NurApi.LOG_DATA);
            NurCapabilities.I.Nur = hNur
        Catch ex As Exception
            MessageBox.Show(ex.Message, Program.appName)
            Me.Close()
            Return
        End Try

        ' Set NurApi for UserControls
        nurLog.SetNurApi(hNur)
        nurInfo.SetNurApi(hNur)
        nurInventory.SetNurApi(hNur)
        nurWriter.SetNurApi(hNur)
        nurLocator.SetNurApi(hNur)
        nurSettings.SetNurApi(hNur)
        nurNxp.SetNurApi(hNur)

        Try
            ' Configure Scan & Trigger -buttons
            HHScanButton.ConfigureScanButtons(DirectCast(NordicId.VK.SCAN, Keys))
            HHScanButton.ScanButtonMode = HHScanButton.SCANMODE.NONE

        Catch generatedExceptionName As Exception
        End Try
    End Sub

    ''' <summary>
    ''' Handles the Load event of the Form1 control.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="EventArgs" /> instance containing the event data.</param>
    Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
        ' Connect Nur -module
        hNur.ConnectIntegratedReader()
    End Sub

    ''' <summary>
    ''' Handles the Closing event of the Form1 control.
    ''' </summary>
    ''' <param name="sender">The source of the event.</param>
    ''' <param name="e">The <see cref="CancelEventArgs" /> instance containing the event data.</param>
    Private Sub Form1_Closing(ByVal sender As Object, ByVal e As CancelEventArgs)
        ' Restore Scan & Trigger -button configurations
        HHScanButton.RestoreScanButtons()

        ' Stop locationg thread if running
        nurLocator.StopLocating()

        ' Dispose NurApi
        If hNur IsNot Nothing Then
            If hNur.IsConnected() Then
                hNur.Disconnect()
            End If
            hNur.Dispose()
        End If
        Fullscreen.SetFullScreen(False)
    End Sub

    Private Sub Form1_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
        ' Make sure that this.KeyPreview is turned on.
        Select Case e.KeyCode
            Case DirectCast(NordicId.VK.SCAN, Keys)
                ' Jump to inventory -tab and start / stop inventory
                tabControl1.SelectedIndex = tabControl1.TabPages.IndexOf(inventoryTab)
                nurInventory.startStopInventoryBtn_Click(Me, EventArgs.Empty)
                e.Handled = True
                Exit Select
        End Select
    End Sub
End Class
